### ISSUE TITLE
Extrude Separate produces extra vertices needing remove doubles?

### ISSUE THREAD

**Issue by enzyme69** [nortikin/sverchok#1552]
I made this Sverchok Spinner using Extrude, Subdiv and Smooth. It is looking alright, but throughout the process I noticed a couple of gotchas.

GIST:
https://gist.github.com/d1ba5bd11ab3e5b5e0296a85f6fecef3

![screen shot 2017-04-21 at 11 31 39](https://cloud.githubusercontent.com/assets/3952687/25258938/538300a2-2686-11e7-970c-f5b14217143b.png)

- Often after Extrude Separate Face, I need to do Remove Doubles. Even if I do List Mask Face
- When extruding the middle part of the Spinner, I found Extrude Region normal does not quite work as expected. Scale works though.
- What would be nice is to be able to extrude certain faces and then in the next extrude to keep track of those faces. Of course more ideal is to have some kind of Grouping that keeps track of the polygon face index.
- Would be nice to really get polygon face index by selection.. although this is not very procedural in the long run because it is absolute index.
- Polygon Face Mask Selection needs a way to just do List Mask by providing index: 1,5,12, etc I know this can be done already the hard way. Just like if use give 1,5, and we compare with range 0-6, we get False, True, False, False, True, False.

**Comment by zeffii** [comment]
>  Polygon Face Mask Selection needs a way to just do List Mask by providing index: 1,5,12, etc I know this can be done already the hard way. Just like if use give 1,5, and we compare with range 0-6, we get False, True, False, False, True, False.

new topic just for that. I agree https://github.com/nortikin/sverchok/issues/1553

**Comment by zeffii** [comment]
give this issue a better title, it seems to be mostly about `Extrude separate`..  focus on the problem.  The fact that you are making a spinning top is not relevant to the issue.

**Comment by zeffii** [comment]
>  although this is not very procedural in the long run because

that's also a good point and worth emphasis . Just because someone is using nodes, doesn't mean they approach the problem procedurally. When we are simply translating the manual way into nodes, sure... we have some parametric solution... but parametric doesn't necessarily mean procedural.

**Comment by zeffii** [comment]
@enzyme69  , to demonstrate that problem i'm sure you could show it using just a plane with 4 faces. and a much smaller setup

**Comment by portnov** [comment]
Ok, talking about new title of this issue, I actually see the issue too. I can't remember now why I implemented this node this way. It currently applies extrude operation to all faces, and selection mask only decides which of extruded faces to move/scale; other stay in place, so we have duplicated vertices.

I could change the implementation. But, i'm not quite sure: maybe someone relies on current implementation? Maybe I should now add "compatibility mode" parameter (default True)?
